src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -336,51 +261,183 @@ void parseCvrFile(List<CastVoteRecord> castVoteRecords)
     }
   }
 
-  void parseJson(List<CastVoteRecord> castVoteRecords) throws UnrecognizedCandidatesException {
-    // cvrIndex and fileName are used to generate IDs for cvrs
+  private HashMap getCvrContestJson(HashMap cvr, String contestIdToTabulate) {
+    HashMap cvrContestToTabulate = null;
+    String currentSnapshotId = (String) cvr.get("CurrentSnapshotId");
+    ArrayList cvrSnapshots = (ArrayList) cvr.get("CVRSnapshot");
+    HashMap currentSnapshot = null;
+    for (Object snapshotObject : cvrSnapshots) {
+      HashMap snapshot = (HashMap) snapshotObject;
+      if (snapshot.get("@id").equals(currentSnapshotId)) {
+        currentSnapshot = snapshot;
+        break;
+      }
+    }
+    // find the cvr contest in this snapshot
+    ArrayList cvrContests = (ArrayList) currentSnapshot.get("CVRContest");
+    for (Object contestObject : cvrContests) {
+      HashMap cvrContest = (HashMap) contestObject;
+      // filter by contest ID
+      String contestId = (String) cvrContest.get("ContestId");
+      if (contestId.equals(contestIdToTabulate)) {
+        cvrContestToTabulate = cvrContest;
+        break;
+      }
+    }
+    return cvrContestToTabulate;
+  }
+
+  // parse cdf json CastVoteRecordReport into CastVoteRecords and append them to input list
+  void parseJson(List<CastVoteRecord> castVoteRecords)
+      throws CvrParseException, UnrecognizedCandidatesException {
+
+    // static election data
+    HashMap<Object, Object> candidates = new HashMap<>();
+    HashMap<Object, Object> gpuUnits = new HashMap<>();
```

Comment:
This doesn't seem like it's actually used for anything.

Reply:
True - these are for precincts - I just added the parsing code.